core/templates/hash_map.h

```diff
@@ -71,6 +71,7 @@ class HashMap : private Allocator {
 	static constexpr uint32_t MIN_CAPACITY_INDEX = 2; // Use a prime.
 	static constexpr float MAX_OCCUPANCY = 0.75;
 	static constexpr uint32_t EMPTY_HASH = 0;
+	using KV = KeyValue<TKey, TValue>; // Type alias for easier access to KeyValue.
```

Comment:
Particularly useful when you have a typedef such as:

```cpp
typedef HashMap<uint32_t, VertexAttributeBinding> VertexAttributeBindingsMap;
```

as you can then use:

```
for (const VertexAttributeBindingsMap::KV &ky : p_vertex_bindings) {
  // ...
}
```

vs

```cpp
for (const KeyValue<uint32_t, VertexAttributeBinding> &kv : p_vertex_bindings) {
  // ...
}
```

Also, if you change the Key or Value type, these call sites don't need to be updated.